src/backend/tasks_io/datafeeds/datafeed_fms_api.py

```diff
@@ -0,0 +1,320 @@
+# from parsers.fms_api.fms_api_team_avatar_parser import FMSAPITeamAvatarParser
+
+
+class DatafeedFMSAPI:
+
+    EVENT_SHORT_EXCEPTIONS = {
+        "arc": "archimedes",
+        "cars": "carson",
+        "carv": "carver",
+        "cur": "curie",
+        "dal": "daly",
+        "dar": "darwin",
+        "gal": "galileo",
+        "hop": "hopper",
+        "new": "newton",
+        "roe": "roebling",
+        "tes": "tesla",
+        "tur": "turing",
+    }
+
+    SUBDIV_TO_DIV = {  # 2015, 2016
+        "arc": "arte",
+        "cars": "gaca",
+        "carv": "cuca",
+        "cur": "cuca",
+        "gal": "gaca",
+        "hop": "neho",
+        "new": "neho",
+        "tes": "arte",
+    }
+
+    SUBDIV_TO_DIV_2017 = {  # 2017+
+        "arc": "arda",
+        "cars": "cate",
+        "carv": "cane",
+        "cur": "cuda",
+        "dal": "arda",
+        "dar": "cuda",
+        "gal": "garo",
+        "hop": "hotu",
+        "new": "cane",
+        "roe": "garo",
+        "tes": "cate",
+        "tur": "hotu",
+    }
+
+    def __init__(self) -> None:
+        self.api = FRCAPI()
+
+    def get_awards(self, event: Event) -> List[Award]:
+        awards = []
+
+        # 8 subdivisions from 2015+ have awards listed under 4 divisions
+        if event.event_type_enum == EventType.CMP_DIVISION and event.year >= 2015:
+            event_team_keys = EventTeam.query(EventTeam.event == event.key).fetch(
+                keys_only=True
+            )
+            valid_team_nums = { int(etk.id().split("_")[1][3:]) for etk in event_team_keys }
+
+            if event.year >= 2017:
+                division = self.SUBDIV_TO_DIV_2017[event.event_short]
+            else:
+                division = self.SUBDIV_TO_DIV[event.event_short]
+
+            api_awards_response = self.api.awards(
+                event.year, event_code=DatafeedFMSAPI._get_event_short(division)
+            )
+            # TODO: if parsed_awards else [] - this will throw when it's a None
+            awards += self._parse(
+                api_awards_response, FMSAPIAwardsParser(event, valid_team_nums)
+            )
+
+        api_awards_response = self.api.awards(
+            event.year, event_code=DatafeedFMSAPI._get_event_short(event.event_short, event)
+        )
+        # TODO: if parsed_awards else [] - this will throw when it's a None
+        awards += self._parse(api_awards_response, FMSAPIAwardsParser(event))
+
+        return awards
+
+    @classmethod
+    def _get_event_short(self, event_short: str, event: Optional[Event] = None) -> str:
+        # First, check if we've manually set the FRC API key
+        if event and event.first_code:
+            return event.first_code
+
+        # Otherwise, check hard-coded exceptions
+        return DatafeedFMSAPI.EVENT_SHORT_EXCEPTIONS.get(event_short, event_short)
+
+    def _parse(
+        self, response: requests.Response, parser: ParserBase
```

Comment:
- [ ] Don't merge this until we get the storage stuff put in. Doesn't need `sim_time` for now, does need writes.